src/numberparse.rs

```diff
@@ -64,9 +64,8 @@ fn is_not_structural_or_whitespace_or_exponent_or_decimal(c: u8) -> bool {
 #[cfg(all(feature = "swar-number-parsing"))]
 #[cfg_attr(not(feature = "no-inline"), inline)]
 #[allow(clippy::cast_ptr_alignment)]
-fn is_made_of_eight_digits_fast(chars: &[u8]) -> bool {
-    // We know what we're doing right? :P
-    let val: u64 = unsafe { *(chars.as_ptr().cast::<u64>()) };
+fn is_made_of_eight_digits_fast(chars: [u8; 8]) -> bool {
+    let val = u64::from_ne_bytes(chars);
```

Comment:
Why not just
```rust
fn is_made_of_eight_digits_fast(chars: &[u8]) -> bool {
    // Safety: ...
    let val = unsafe { (chars.as_ptr() as *const u64).read_unaligned() };
    // ...
```

Reply:
Would still be incorrect if the given slice has less than 8 bytes.

Reply:
Edited it shortly before your reply to suggest making the whole function unsafe instead.

Reply:
I think it's fine to ensure the function getting always 8 bytes, seems sound and moving soundness requirements a level up 👍

Reply:
Reading through all the chances and considering that we'd be back with a pointer cast, just moved to two different places, I tend to agree with @hkratz. It'd make more sense to mark the function as unsafe and let it deal with the unsafety/pointer cast in a single place than having to reason about it in multiple places.

Reply:
I would reason the other way around. You put the invariant of the function (that you have a buffer of 8 bytes) into the data type. This will make code safer as you explicitly have to make the cast every time you call it.

On places you are absolutely sure, you can still `unwrap_unchecked` to elide the 8 byte check

Reply:
good point as well 👍